extensions/functions_arithmetic.yaml

```diff
@@ -173,6 +173,178 @@ scalar_functions:
           - value: i64
           - value: i64
         return: i64
+  -
+    name: "cos"
+    description: "Get the cosine of a value in radians."
+    impls:
+      - args:
+          - value: i8
+        return: fp64
+      - args:
+          - value: i16
+        return: fp64
+      - args:
+          - value: i32
+        return: fp64
+      - args:
+          - value: i64
+        return: fp64
+      - args:
+          - value: fp32
+        return: fp64
+      - args:
+          - value: fp64
+        return: fp64
+  -
+    name: "sin"
+    description: "Get the sine of a value in radians."
+    impls:
+      - args:
+          - value: i8
+        return: fp64
+      - args:
+          - value: i16
+        return: fp64
+      - args:
+          - value: i32
+        return: fp64
+      - args:
+          - value: i64
+        return: fp64
+      - args:
+          - value: fp32
+        return: fp64
+      - args:
+          - value: fp64
+        return: fp64
+  -
+    name: "tan"
+    description: "Get the tangent of a value in radians."
+    impls:
+      - args:
+          - options: [ SILENT, SATURATE, ERROR ]
```

Comment:
There is some good input from @jvanstraten on this topic (well, related topic at least) here: https://github.com/substrait-io/substrait/pull/230#issuecomment-1170230630

Reply:
Apologies; should have put this in the original PR.  The reasoning for specifying the overflow behaviour options for tangent only was on the basis that the ranges for the returned value for this function are wider than the other ones and so I assumed it'd need it.

My trig is also a bit rusty, but from what I can gather from a bit of a search, in terms of output values:
* range of the cosine and sine functions: [-1, 1]
* range of  tangent: [-infinity, infinity].  
* range of arctangent, arcsine: [-pi/2, pi/2]
* range of arccosine: [0,pi]

Though as @westonpace has said, the approach suggested by @jvanstraten might be more appropriate here.

Reply:
`tan` is technically undefined at `(1/2 + k) pi` for all integer `k`, because that's a division by zero when expressed as `sin(x) / cos(x)`. However, due to pi's transcendental nature, none of those numbers can be exactly represented with floating-point numbers, so `tan(x)` is defined for all finite floating-point values of x. So `tan` only needs the rounding option.

`asin` and `acos` do have domain errors; they are undefined for `x < -1` and `x > 1`. So they should be defined however `sqrt` ends up being defined. `atan2` is also a bit weird when `0, 0` is passed, since it's basically `atan(y / x)` (with some boundary conditions to deal with `x = 0` when `y != 0` and to get all quadrants to behave nicely), so I guess it should get the same treatment.

The other trig functions are defined for all finite real numbers, and thus only need rounding options.

Reply:
> due to pi's transcendental nature

Especially when topped with ice cream

Reply:
As the conversation regarding those other options now appears to be resolved, have updated the options here in line with those suggestions.